test/parallel/test-http-mutable-headers.js

```diff
@@ -79,8 +79,7 @@ const s = http.createServer(common.mustCall((req, res) => {
         'set-cookie': cookies,
         'x-test-array-header': arrayValues
       });
-      // eslint-disable-next-line no-restricted-properties
-      assert.deepEqual(headersCopy['set-cookie'], cookies);
+      assert.deepStrictEqual(headersCopy['set-cookie'], cookies);
```

Comment:
In my books it's not an Object if `headersCopy instanceof Object === false` 🤷‍♂️ 
Ref: #12885